lib/src/Index.cpp

```diff
@@ -35,12 +35,14 @@ using namespace utility;
 using namespace web::json;
 
 Index::Index(
-		IndexType kind,
-		const utility::string_t& data_type,
-		int string_precision)
-	: kind_(kind)
-	, data_type_(data_type)
+		IndexType index_type,
+		int numeric_precision,
+		int string_precision,
+		string_t path)
+	: index_type_(index_type)
+	, numeric_precision_(numeric_precision)

```

Comment:
So, I made mistake in the past calling this "string_precision", I see now. This lead you to add new precision (numeric), but lookign at REST spec here, I think we are better with having 1 precision called "precision" (just renaming existing string_precision to precision). What do you think?

Reply:
I will. But then - it is better that you do not modify these lines, because it is extra work that will be removed anyway. So, maybe please don't modify these indexes/IndexPath at all. After we push your changes (for 429/retry) and after I make all precisiont to be named precision, you see if there is anything else missing for you. How about that?